A103-xds-composite-filter.md

```diff
@@ -0,0 +1,171 @@
+  Specifies a unified matcher tree indicating the config of the filter
+  to use.  The actions in this tree must be one of two types:
+  - [`envoy.extensions.filters.common.matcher.action.v3.SkipFilter`](https://github.com/envoyproxy/envoy/blob/0685d7bf568485eb112df2a9c73248cb8bfc1c37/api/envoy/extensions/filters/common/matcher/action/v3/skip_action.proto#L24):
+    This indicates that no filter will be executed.
+  - [`envoy.extensions.filters.http.composite.v3.ExecuteFilterAction`](https://github.com/envoyproxy/envoy/blob/0685d7bf568485eb112df2a9c73248cb8bfc1c37/api/envoy/extensions/filters/http/composite/v3/composite.proto#L49):
+    This indicates which filter should be executed.  Within it:
+    - [typed_config](https://github.com/envoyproxy/envoy/blob/0685d7bf568485eb112df2a9c73248cb8bfc1c37/api/envoy/extensions/filters/http/composite/v3/composite.proto#L54C39-L54C51):
+      The filter to configure.  It will be validated using the xDS HTTP filter
+      registry, just as the top-level filters in the HTTP connection manager
+      config are.  This field is ignored if `filter_chain` is set.  It
+      is an error if neither `typed_config` nor `filter_chain` are set.
+    - [dynamic_config](https://github.com/envoyproxy/envoy/blob/0685d7bf568485eb112df2a9c73248cb8bfc1c37/api/envoy/extensions/filters/http/composite/v3/composite.proto#L59):
+      This field will be ignored for now, since gRPC does not currently
+      support ECDS.  Support for ECDS will be added in a subsequent gRFC.
+    - filter_chain (new field added in
+      https://github.com/envoyproxy/envoy/pull/40885): This specifies a
+      chain of filters to call, in order.  Each filter in the chain will
+      be validated using the xDS HTTP filter registry, just as the
+      top-level filters in the HTTP connection manager config are.
+      If set, the `typed_config` field is ignored.  It is an error if
+      neither `typed_config` nor `filter_chain` are set.
+    - [sample_percent](https://github.com/envoyproxy/envoy/blob/0685d7bf568485eb112df2a9c73248cb8bfc1c37/api/envoy/extensions/filters/http/composite/v3/composite.proto#L69C43-L69C57):
+      - Optional; if unset, the specified filter(s) are always executed.
+        If set, for each RPC, a random number will be generated between
+        0 and 100, and if that number is less than the specified
+        threshold, the specified filter(s) will be executed.
+        Within this field:
+        - [default_value](https://github.com/envoyproxy/envoy/blob/cdd19052348f7f6d85910605d957ba4fe0538aec/api/envoy/config/core/v3/base.proto#L648):
+          This field must be present. The configured value will be capped at
+          100%.
+        - runtime_key: This field will be ignored, since gRPC does not
+          have a runtime system.
+- matcher: gRPC will not support this deprecated field.
+
+We will also support per-route overrides via the
+[`envoy.extensions.common.matching.v3.ExtensionWithMatcherPerRoute`
```

Comment:
So this means we can have filters defined wholly in the RouteConfiguration? The Listener would no longer enumerate all the filters being used.